lib/Semantics/check-namelist.cpp

```diff
@@ -0,0 +1,40 @@
+//===-- lib/Semantics/check-namelist.cpp ----------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "check-namelist.h"
+
+namespace Fortran::semantics {
+
+void NamelistChecker::Leave(const parser::NamelistStmt &nmlStmt) {
+  for (auto &x : nmlStmt.v) {
+    if (auto *nml{std::get<parser::Name>(x.t).symbol}) {
+      for (const auto &nmlObjName : std::get<std::list<parser::Name>>(x.t)) {
+        auto *nmlObjSymbol{nmlObjName.symbol};
```

Comment:
I think it's helpful to use `const auto` when you can rather than just `auto`. That applies to `x`, `nml`, and `nmlObjSymbol`.